### PR TITLE
Issue #239: remove checkstyle source zip from git and use download

### PR DESCRIPTION
**Pull request by mickroll**
downloads from https://github.com/checkstyle/checkstyle/archive/checkstyle-${checkstyle.version}.zip

### CODE REVIEW

**Comment by mickroll** [comment]
Hint: right now it seems impossible to also remove the *-all.jar (right next to the zip file). See #239

**Comment by romani** [comment]
@mickroll , please rebase to resolve conflict.

**Comment by mickroll** [comment]
Rebased, squashed and updated `checkstyle.version` to fit version present in master branch.

File: net.sf.eclipsecs.checkstyle/pom.xml [hunk 1]
```diff
@@ -9,8 +9,42 @@
...
+                <artifactId>download-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>download-checkstyle-jar</id>
+                        <phase>initialize</phase>
+                        <goals>
+                            <goal>wget</goal>
+                        </goals>
+                        <configuration>
+                            <skip>true</skip><!-- jar has not the same content as the one present. maybe it is built manually!? -->
```
Comment (Calixte) [inline net.sf.eclipsecs.checkstyle/pom.xml:29]: Can you only take care of the zip file in this PR? I don't think downloading the jar dependency using download-maven-plugin will ever work.

File: pom.xml [hunk 2]
```diff
@@ -121,6 +121,11 @@
                     <artifactId>maven-antrun-plugin</artifactId>
                     <version>1.8</version>
                 </plugin>
+                <plugin>
+					<groupId>com.googlecode.maven-download-plugin</groupId>
```
Comment (Calixte) [inline pom.xml:125]: This file is indented using spaces and it seems that you used tabs.

File: net.sf.eclipsecs.checkstyle/pom.xml (hunk 1 above)
Comment (mickroll) [reply net.sf.eclipsecs.checkstyle/pom.xml]: Done.

File: pom.xml (hunk 2 above)
Comment (mickroll) [reply pom.xml]: good catch. converted to spaces.

**Pull request merged**